pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go

```diff
@@ -195,11 +196,21 @@ func (c *UpgradeBackupController) ensureRecentBackup(ctx context.Context, cluste
 	case backupPod.Status.Phase == corev1.PodSucceeded:
 		backupCondition := configv1helpers.FindStatusCondition(clusterOperatorStatus.Conditions, backupConditionType)
 		if backupCondition.Reason != backupSuccess {
```

Comment:
Should this not be be true for when backup is successful as well:

```suggestion
		if backupCondition.Reason == backupSuccess {
```

otherwise, we will never update the backupDirName once backup is successful first time?

Reply:
IIUC we might not need to update the backupDirName ever again in the status if the backup pod has already succeeded, and we've already once set the backup condition reason to `UpgradeBackupSuccessful`.

We'd only need to rewrite the backupDirName for a new backup, that results in a new backupdir.

And if ([for whatever reason](https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go#L160)), a new backup is initiated , then the backupCondtion would be reset to `UpgradeBackupInProgress`. So here it would just update it again.
https://github.com/openshift/cluster-etcd-operator/blob/2ef42e68e7acdfd7626e481f196d0f1705556c34/pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go#L175-L184

Reply:
Thank you! Makes sense to me!